analyzer/config/helm/helm.go

```diff
@@ -0,0 +1,133 @@
+package helm
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"errors"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	"github.com/aquasecurity/fanal/types"
+	"golang.org/x/xerrors"
+)
+
+const version = 1
+
+type ConfigAnalyzer struct {
+	filePattern *regexp.Regexp
+}
+
+func NewConfigAnalyzer(filePattern *regexp.Regexp) ConfigAnalyzer {
+	return ConfigAnalyzer{
+		filePattern: filePattern,
+	}
+}
+
+func (a ConfigAnalyzer) Analyze(_ context.Context, input analyzer.AnalysisInput) (*analyzer.AnalysisResult, error) {
+	b, err := io.ReadAll(input.Content)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to read %s: %w", input.FilePath, err)
+	}
+
+	return &analyzer.AnalysisResult{
+		Files: map[types.HandlerType][]types.File{
+			// it will be passed to misconfig post handler
+			types.MisconfPostHandler: {
+				{
+					Type:    types.Helm,
+					Path:    input.FilePath,
+					Content: b,
+				},
+			},
+		},
+	}, nil
+}
+
+func (a ConfigAnalyzer) Required(filePath string, _ os.FileInfo, readerOpener analyzer.Opener) bool {
+	if a.filePattern != nil && a.filePattern.MatchString(filePath) {
+		return true
+	}
+
+	if isArchive(filePath) {
+		reader, err := readerOpener()
+		defer func() { _ = reader.Close() }()
+		if err != nil {
+			return false
+		}
+		return isHelmChart(filePath, reader)
```

Comment:
Can we do this detection in `Analyze()`? Then, we don't have to pass `opener` to `Required`.

Reply:
Yeah, I appreciate not ideal to pass in the opener however this was so we cool look inside the tarballs and see if they were helm charts. I'm trying to avoid adding any found `tar`, `tgz`, or `tar.gz` that is found being added into the scanned file system.

I can look to move it into the Anaylze but I'm concerned about adding random very large tarballs that aren't needed.

Reply:
It might be okay if I change https://github.com/aquasecurity/fanal/blob/main/analyzer/analyzer.go#L311-L322 to handle `nil` being returned and not try an merge. then the `IsHelmChart` logic can move. 

If this is preferable, I'll do that

Reply:
`opener` returns *os.File.
https://github.com/aquasecurity/fanal/blob/ca3bf967cb0d97a8a9939ffd0a702c2fa5fa904b/walker/fs.go#L66-L70

And `isHelmChart` processes tar.gz streamingly with gzip.NewReader and tar.NewReader. I believe we don't load the tarball into memory unless we detect it as Helm chart.
https://github.com/aquasecurity/fanal/blob/fbb8cf9d9e598d2ee31fe513efe38282ce8fe5ac/analyzer/config/helm/helm.go#L90-L116

 If your 10GB photo tarball includes `Chart.yaml`, it will be problematic.
https://github.com/aquasecurity/fanal/blob/fbb8cf9d9e598d2ee31fe513efe38282ce8fe5ac/analyzer/config/helm/helm.go#L111-L113

Reply:
But we can skip too large tarball anyway because it is probably not Helm chart.

Reply:
added an arbitrary size gate of 200mb and moved the IsHelm call into the Analyze function. 

A lot less intrusive change now